internal/provider/database/resource.go

```diff
@@ -64,7 +76,44 @@ func (state *DatabaseResourceModel) Read(ctx context.Context, client openapi.Cli
 	return helper.ParseResponse(resp, state)
 }
 
-func (state *DatabaseResourceModel) Update(ctx context.Context, client openapi.ClientInterface) error {
+const (
+	DBA_PASSWORD_CHANGE_UNSUPPORTED_MSG = "Configured DBA password was changed and the server does not support updating the DBA password. " +
+		"Revert the configured DBA password to the value in the Terraform state and retry."
+)
+
+func IsDbaPasswordUpdateUnsupportedError(resp *http.Response, err error) bool {
+	if err != nil {
+		// "404 Not Found" is returned with no "detail" message if /dbaPassword	sub-resource is not supported
```

Comment:
Nit: Can we mention that in future versions the `UNKNOWN_REQUEST` code will be returned, however, this change appears in the same version as the `/dbaPassword` sub-resource so we can't use it here?

Reply:
Right, if `UNKNOWN_REQUEST` appears, then we would expect `/dbaPassword` to be supported.